keyboards/ergodox_ez/keymaps/rmw/keymap-mw2.c

```diff
@@ -0,0 +1,387 @@
+    SS_LSFT(SS_LGUI(SS_TAP(X_RIGHT)))      // MAC
+  }
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    // dynamically generate these.
+    case EPRM:
+      if (record->event.pressed) {
+        eeconfig_init();
+      }
+      return false;
+      break;
+    case VRSN:
+      if (record->event.pressed) {
+        SEND_STRING (QMK_KEYBOARD "/" QMK_KEYMAP " @ " QMK_VERSION);
+      }
+      return false;
+      break;
+    case FORM_GET:
+      if (record->event.pressed) {
+        //click KC_MS_BTN1
+        //register_code16(KC_BTN1);
+        //unregister_code16(KC_BTN1);
+        tap_code16(KC_MS_BTN1);
+        if (mac) {
+          SEND_STRING(SS_LGUI("a"));
+          SEND_STRING(SS_LGUI("c"));
+        }
+        else {
+          SEND_STRING(SS_LCTRL("a"));
+          SEND_STRING(SS_LCTRL("c"));
+        }
+      }
+      return false;
+      break;
+    case FORM_PUT:
+      if (record->event.pressed) {
+        tap_code16(KC_MS_BTN1);
+        if (mac) {
+          SEND_STRING(SS_LGUI("a"));
+          SEND_STRING(SS_LGUI("v"));
+        }
+        else {
+          SEND_STRING(SS_LCTRL("a"));
+          SEND_STRING(SS_LCTRL("v"));
+        }
+      }
+      return false;
+      break;
+
+    case MAC_WIN:
+      if (record->event.pressed) {
+        mac = !mac;
+        //SEND_STRING("MAC_WIN");
+      }
+      return false;
+      break;
+
+    case MVW_LEFT ... SEL_END: 
+      if (record->event.pressed) {
+        uint16_t index = keycode - MVW_LEFT;
+        // Choose based on which key was pressed and whether OS is Win or Mac.
+        if (mac){
+          send_string(macwin_codes[index][1]);
+          // something here needs to clear the one-shot layer.
+        }
+        else {
+          send_string(macwin_codes[index][0]);
+          // something here needs to clear the one-shot layer.
+        }
+        return false;
+      }
+      break;
+  }
+    return true;
+}
+
+void matrix_scan_user(void) {
+    uint8_t layer = biton32(layer_state);
+    ergodox_board_led_off();
+    ergodox_right_led_1_off();
+    ergodox_right_led_2_off();
+    ergodox_right_led_3_off();
+    switch (layer) {
+        case 1:
+            ergodox_right_led_1_on();
+            break;
+        case 2:
+            ergodox_right_led_2_on();
+            break;
+        case 3:
+            ergodox_right_led_3_on();
+            break;
+        case 4:
+            ergodox_right_led_1_on();
+            ergodox_right_led_2_on();
+            break;
+        case 5:
+            ergodox_right_led_1_on();
+            ergodox_right_led_3_on();
+            break;
+        case 6:
+            ergodox_right_led_2_on();
+            ergodox_right_led_3_on();
+            break;
+        case 7:
+            ergodox_right_led_1_on();
+            ergodox_right_led_2_on();
+            ergodox_right_led_3_on();
+            break;
+        default:
+            break;
+    }
```

Comment:
Huh. This was just reused from the default ergodox_ez keymap from back when it was ergodox/ez. I'll have a look.

Reply:
I'm having trouble getting this to work right with MOs and LTs.... It seems to activate the light change only after I release the key I was holding to temporarily dip into the layer, so my default layer glows with whatever the last non-default layer I dipped into was. I've updated the function to use get_highest_layer rather than the biton32, but it seems it may have to just live in the matrix_scan.